sycl/include/sycl/ext/intel/experimental/esimd/detail/memory_intrin.hpp

```diff
@@ -347,8 +755,35 @@ __esimd_lsc_load_stateless(__ESIMD_DNS::simd_mask_storage_t<N> pred,
     ;
 #else  // __SYCL_DEVICE_ONLY__
 {
-  __ESIMD_UNSUPPORTED_ON_HOST;
-  return 0;
+  // TODO : Support AddressScale, ImmOffset
+  static_assert(AddressScale == 1);
+  static_assert(ImmOffset == 0);
+  static_assert(DS != __ESIMD_ENS::lsc_data_size::u16u32h);
+
+  constexpr uint MASK = loadstoreAlignMask<Ty, VS, DS, N>();
+  __ESIMD_DNS::vector_type_t<Ty, N * __ESIMD_EDNS::to_int<VS>()> Output = 0;
+
+  constexpr int ElemCount = __ESIMD_EDNS::to_int<VS>();
```

Comment:
I think N's definition `/// @tparam N is the number of channels (platform dependent).` is wrong here and in many other places in this file. This is actually the simd size - the number of chunks of VS*sizeof(T) size. If we take rgba load/strore as reference, then VS would be the number of channels, not N. So `ElemCount` is actually the number of channels. It would be great if you could change `N`'s definition to something like "the SIMD size of operation (the number of addresses to access)".

Reply:
Comment updated.